src/kademlia/traversal_algorithm.cpp

```diff
@@ -593,6 +593,24 @@ void traversal_algorithm::status(dht_lookup& l)
 	l.last_sent = last_sent;
 }
 
+void look_for_nodes(node const& node, bdecode_node const& r, std::function<void(const node_endpoint&)> f)
```

Comment:
I suppose you would need the ``protocol`` object as well. perhaps the nodes key could be derived from it too..